lib/backend/k8s/conversion.go

```diff
@@ -382,18 +404,42 @@ func (c Converter) k8sProtocolToCalico(protocol *extensions.Protocol) *numorstri
 	return nil
 }
 
-func (c Converter) k8sPeerToCalicoSelector(peer extensions.NetworkPolicyPeer, ns string) string {
+func (c Converter) k8sPeerToCalicoSrcFields(peer extensions.NetworkPolicyPeer, ns string) (string, []*cnet.IPNet, []*cnet.IPNet) {
 	// Determine the source selector for the rule.
 	// Only one of PodSelector / NamespaceSelector can be defined.
 	if peer.PodSelector != nil {
-		return c.k8sSelectorToCalico(peer.PodSelector, &ns)
+		return c.k8sSelectorToCalico(peer.PodSelector, &ns), nil, nil
 	}
 	if peer.NamespaceSelector != nil {
-		return c.k8sSelectorToCalico(peer.NamespaceSelector, nil)
+		return c.k8sSelectorToCalico(peer.NamespaceSelector, nil), nil, nil
+	}
+	if peer.IPBlock != nil {
+		// Convert the CIDR to include.
+		nets := []*cnet.IPNet{}
+		_, ipNet, err := cnet.ParseCIDR(peer.IPBlock.CIDR)
+		if err != nil {
+			log.WithField("cidr", peer.IPBlock.CIDR).WithError(err).Error("Failed to parse CIDR")
+			goto parseError
```

Comment:
Does this `goto` buy us anything over just returning here?  Seems like that would be easier to follow.

Reply:
Well then I have to repeat the 'null' return tuple (i.e. `"", nil, nil`)  I thought it could be worth a goto to avoid that.

Reply:
I guess I could set up the null return values at the top of the func, and then just return here...